src/Swarm/TUI/Controller.hs

```diff
@@ -590,25 +612,49 @@ handleREPLEvent s (Key V.KUp) =
   continue $ s & adjReplHistIndex Older
 handleREPLEvent s (Key V.KDown) =
   continue $ s & adjReplHistIndex Newer
+handleREPLEvent s (ControlKey 'r') = continue $
+  case s ^. uiState . uiReplForm . to formState of
+    CmdPrompt uinput ->
+      let newform = mkReplForm $ SearchPrompt uinput (s ^. uiState . uiReplHistory)
+       in s & uiState . uiReplForm .~ newform
+    SearchPrompt ftext rh -> case toList $ rh ^. replSeq of
+      [] -> s
+      _ : rhis ->
```

Comment:
Actually, I think the problem is that if you type many times the same command, It will appear many times in the history and you al filtering out once at time. 

Using `Data.Seq` directly is by far a better solution... you still have the problem of repeated elements in history I'll work that around